### PR TITLE
Split big index.js file into multiple smaller files

### PR DESCRIPTION
**Pull request by yuvipanda**
- Follows the plan laid out in
https://github.com/jupyterhub/binderhub/issues/777
- Only non-trivial move is to cleanup the `_config` call to
  use the more modern 'fetch' API rather than the older
  XMLHttpRequest API

Fixes https://github.com/jupyterhub/binderhub/issues/777

A screen recording of me testing this is available in
https://github.com/jupyterhub/binderhub/pull/1758

### CODE REVIEW

**Comment by choldgraf** [comment]
omg is it finally happening?!

File: binderhub/static/js/src/repo.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const configUrl = BASE_URL + "_config";
+    fetch(configUrl).then(resp => {
+      resp.json().then(data => {
+        configDict = data
+        setLabels();
+      });
+    })
+  } else {
+    setLabels();
+
```
Comment (consideRatio) [inline binderhub/static/js/src/repo.js:41]:
```suggestion
```

File: binderhub/static/js/src/urls.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { makeBadgeMarkup } from './badge';
+import { getBuildFormValues } from './form';
+import { BADGE_BASE_URL, BASE_URL } from './constants';
+
+ function v2url(providerPrefix, repository, ref, path, pathType) {
```
Comment (consideRatio) [inline binderhub/static/js/src/urls.js:5]:
```suggestion
function v2url(providerPrefix, repository, ref, path, pathType) {
```

File: binderhub/static/js/index.js [hunk 3]
```diff
@@ -29,121 +26,12 @@ import 'bootstrap/dist/css/bootstrap.min.css';
...
-  $("label[for=repository]").text(text);
-}
-
-function updateRepoText() {
-  if (Object.keys(config_dict).length === 0){
-    loadConfig(function(res) {
-      config_dict = JSON.parse(res);
-      setLabels();
-    });
-  } else {
```
Comment (consideRatio) [inline binderhub/static/js/index.js:87]: This section previously looked like:

```js
  if (Object.keys(configDict).length === 0) {
    const configUrl = BASE_URL + "_config";
    fetch(configUrl).then(resp => {
      resp.json().then(data => {
        configDict = data
        setLabels();
      });
    })
  }
```

The loadConfig function now in repo.js is unchanged, looking like below with XMLHttpRequest being used still. I understand it as you've inlined the function into updateRepoText, and that updateRepoText was the only place where loadConfig was used, so I figure we should delete it.

```js
function loadConfig(callback) {
  const req = new XMLHttpRequest();
  req.onreadystatechange = function() {
    if (req.readyState == 4 && req.status == 200)
      callback(req.responseText)
  };
  req.open('GET', BASE_URL + "_config", true);
  req.send(null);
}
```

File: binderhub/static/js/src/repo.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  const req = new XMLHttpRequest();
+  req.onreadystatechange = function() {
+    if (req.readyState == 4 && req.status == 200)
+      callback(req.responseText)
+  };
+  req.open('GET', BASE_URL + "_config", true);
+  req.send(null);
+}
+
+
```
Comment (consideRatio) [inline binderhub/static/js/src/repo.js:15]:
```suggestion
```

I think this is removable at this point, motivated in another comment

**Review by consideRatio — COMMENTED** [review]
Thank you for working this @yuvipanda!!!

LGTM except a few (I think) easy-to-resolve comments!

File: binderhub/static/js/index.js (hunk 3 above)
Comment (yuvipanda) [reply binderhub/static/js/index.js:87]: Good catch. done

**Comment by yuvipanda** [comment]
Thanks @consideRatio. I've applied your suggested changes.

I think after this gets merged, I'll enable prettier for JS in this repo as well, so that should handle all other formatting inconsistencies.

File: binderhub/static/js/src/repo.js [hunk 5]
```diff
@@ -1,18 +1,5 @@
...
-      callback(req.responseText)
-  };
-  req.open('GET', BASE_URL + "_config", true);
-  req.send(null);
-}
-
-
 function setLabels() {
   const provider = $("#provider_prefix").val();
   const text = configDict[provider]["text"];
```
Comment (consideRatio) [inline binderhub/static/js/src/repo.js:5]: This commit removed `configDict = {}`, but that is seen referenced via `setLabels`. I think something is going to go wrong

Comment (yuvipanda) [reply binderhub/static/js/src/repo.js:7]: Ah yes, fixed. I thought I had gotten rid of it, but forgotten I hadn't actually gotten rid of all the places it's used :) coming in a different PR!

Comment (yuvipanda) [reply binderhub/static/js/src/repo.js:7]: @consideRatio also I'm not sure if you caught it or the linter did, but eslint would've caught this one as well. So yay for more linting and testing.

**Comment by consideRatio** [comment]
Wieeee thank you @yuvipanda for taking the time to this!!! :heart: :tada:

**Pull request merged**